p4-16/psa/PSA.mdk

```diff
@@ -421,41 +447,40 @@ error to instantiate the `clone` extern in the P4 control or parser
 block.
 
 A PSA implementation provides two configuration metadata, `clone`
-and `clone_spec`, to the PRE to control the cloning mechanism. The
-`clone` enables/disables the cloning mechanism. If the `clone`
-bit is set, the clone mechanism generates a cloned packet with the
-optional attached metadata. If the `clone` bit is unset or
-uninitialized, the clone mechanism is disabled and a cloned packet is
+and `clone_port`, to the PRE to control the cloning mechanism. When
+the `clone` bit is set, the clone mechanism generates a cloned packet with the
+optional attached metadata. When the `clone` bit is unset or
+uninitialized, cloning is disabled and a cloned packet is
 not generated even if the `emit` method is invoked in the deparser.
-The `clone_spec` control the destination of the cloned packet. A
+The `clone_port` controls the destination of the cloned packet. A
 common use case is to the send the cloned packet to the control CPU,
-in which case the `clone_spec` should be set to the value that
+in which case the `clone_port` should be set to the value that
 represents the control CPU port.
 
 The PSA specifies four types of cloning, with the packet sourced from
 different points in the pipeline and sent back to ingress or to the
-buffering queue in the egress. 
+buffering queue in the egress. (TBD: Which are the four??)
 
 ```
-extern clone {
-  /// Write @hdr into the ingress/egress clone engine.
-  /// @T can be a header type, a header stack, a header union, or a struct
-  /// containing fields with such types.
-  void emit<T>(in T hdr);
-}
+[INCLUDE=psa.p4:Cloning_methods]
```

Comment:
If the restrictions on clone options that @hanw is proposing are kept (see this PR, which maybe should be merged in before this one? https://github.com/p4lang/p4-spec/pull/406 ), then these enum's in psa.p4 will be unnecessary, and the PR I linked proposes deleting those enum types completely.

Reply:
Let's merge andy's PR first, then this one. I will follow up with another PR to clean up the clone section today.